Chapter 13: Constant Time Lookup of Next Node

christopher-besch opened the pull request:
Isn't the lookup time of the next node constant instead of logarithmic?
Current version:
> Using a priority queue, the next node to be processed can be retrieved in logarithmic time.

Referring to cppreference.com:
> A priority queue [...] provides *constant time lookup* of the largest (by default) element, at the expense of logarithmic insertion and extraction.

Referring to Page 43:
> A priority queue maintains a set of elements [...] and retrieval takes O(1) time.

christopher-besch committed:
Chapter 13: Constant Time Lookup of Next Node

Referring to cppreference.com:
> A priority queue [...] provides *constant time lookup* of the largest (by default) element, at the expense of logarithmic insertion and extraction.
Referring to Page 43:
> A priority queue maintains a set of elements [...] and retrieval takes O(1) time.

apronchenkov commented:
I think it depends on the interpretation of "retrieved" -- if it includes taking out the node from the queue, it's logarithmic time for heap.

Maybe the fix should be in the chapter 4, something like: "Insertion and removal take O(logn) time,
and finding the maximum takes O(1) time.

christopher-besch committed:
Update chapter04.tex

christopher-besch committed:
Update chapter13.tex

christopher-besch commented:
That makes sense; thanks for clearing that up for me.

I've implemented the suggested changes and replaced `retrieve` with `lookup`. What do you say?
